assets/js/modules/analytics/datastore/settings.test.js

```diff
@@ -656,6 +767,79 @@ describe( 'modules/analytics settings', () => {
 				).toThrow( INVARIANT_INVALID_PROFILE_SELECTION );
 			} );
 
+			it( 'does not require a valid propertyID when the `ga4Reporting` feature flag is enabled', () => {
+				enabledFeatures.add( 'ga4Reporting' );
+
+				registry.dispatch( MODULES_ANALYTICS ).setSettings( {
+					...validSettings,
+					accountID: '12345',
+					propertyID: null,
+				} );
+
+				registry
+					.dispatch( MODULES_ANALYTICS )
+					.receiveGetExistingTag( tagWithPermission.propertyID );
+
+				expect(
+					registry.select( MODULES_ANALYTICS ).canSubmitChanges()
+				).toBe( true );
+
+				expect( () =>
+					registry
+						.select( MODULES_ANALYTICS )
+						.__dangerousCanSubmitChanges()
+				).not.toThrow( INVARIANT_INVALID_PROPERTY_SELECTION );
```

Comment:
This assertion should be unnecessary because `canSubmitChanges` returning `true` already means that nothing is thrown.

Reply:
Yes, but I'm checking it explicitly that it doesn't throw the specific error. I can remove it; however, we do have a similar assertion:
https://github.com/google/site-kit-wp/blob/6683e33a748075172956102cca4215ecfd6fb90a/assets/js/modules/analytics/datastore/settings.test.js#L886-L894

Reply:
I have removed this assertion 👍

Reply:
Thanks, I'm not sure why that was there before – if `canSubmitChanges` returns `true`, it's guaranteed that the dangerous version won't throw because they use the same validation internally. https://github.com/google/site-kit-wp/blob/4a9611e7538d3172ae0a7c82638e5b9e6e2b34f8/assets/js/googlesitekit/modules/create-submit-changes-store.js#L146-L153

Reply:
Makes sense 👍